lib/JsonpMainTemplatePlugin.js

```diff
@@ -2,205 +2,192 @@
+					}
+				})};`,
+				"var timeout = setTimeout(onScriptComplete, " + chunkLoadTimeout + ");",
+				"script.onerror = script.onload = onScriptComplete;",
+				"function onScriptComplete() {",
+				this.indent([
+					"// avoid mem leaks in IE.",
+					"script.onerror = script.onload = null;",
+					"clearTimeout(timeout);",
+					"var chunk = installedChunks[chunkId];",
+					"if(chunk !== 0) {",
+					this.indent([
+						"if(chunk) chunk[1](new Error('Loading chunk ' + chunkId + ' failed.'));",
+						"installedChunks[chunkId] = undefined;"
+					]),
+					"}"
+				]),
+				"};",
 			]);
-		}
-		return source;
-	});
-	mainTemplate.plugin("jsonp-script", function(_, chunk, hash) {
-		var chunkFilename = this.outputOptions.chunkFilename;
-		var chunkMaps = chunk.getChunkMaps();
-		var crossOriginLoading = this.outputOptions.crossOriginLoading;
-		var chunkLoadTimeout = this.outputOptions.chunkLoadTimeout || 120000;
-		return this.asString([
-			"var script = document.createElement('script');",
-			"script.type = 'text/javascript';",
-			"script.charset = 'utf-8';",
-			"script.async = true;",
-			"script.timeout = " + chunkLoadTimeout + ";",
-			crossOriginLoading ? "script.crossOrigin = '" + crossOriginLoading + "';" : "",
-			"if (" + this.requireFn + ".nc) {",
-			this.indent("script.setAttribute(\"nonce\", " + this.requireFn + ".nc);"),
-			"}",
-			"script.src = " + this.requireFn + ".p + " +
-			this.applyPluginsWaterfall("asset-path", JSON.stringify(chunkFilename), {
-				hash: "\" + " + this.renderCurrentHashCode(hash) + " + \"",
-				hashWithLength: function(length) {
-					return "\" + " + this.renderCurrentHashCode(hash, length) + " + \"";
-				}.bind(this),
-				chunk: {
-					id: "\" + chunkId + \"",
-					hash: "\" + " + JSON.stringify(chunkMaps.hash) + "[chunkId] + \"",
-					hashWithLength: function(length) {
-						var shortChunkHashMap = {};
-						Object.keys(chunkMaps.hash).forEach(function(chunkId) {
-							if(typeof chunkMaps.hash[chunkId] === "string")
-								shortChunkHashMap[chunkId] = chunkMaps.hash[chunkId].substr(0, length);
-						});
-						return "\" + " + JSON.stringify(shortChunkHashMap) + "[chunkId] + \"";
-					},
-					name: "\" + (" + JSON.stringify(chunkMaps.name) + "[chunkId]||chunkId) + \""
-				}
-			}) + ";",
-			"var timeout = setTimeout(onScriptComplete, " + chunkLoadTimeout + ");",
-			"script.onerror = script.onload = onScriptComplete;",
-			"function onScriptComplete() {",
-			this.indent([
-				"// avoid mem leaks in IE.",
-				"script.onerror = script.onload = null;",
-				"clearTimeout(timeout);",
-				"var chunk = installedChunks[chunkId];",
-				"if(chunk !== 0) {",
+		});
+		mainTemplate.plugin("require-ensure", function(_, chunk, hash) {
+			return this.asString([
+				"if(installedChunks[chunkId] === 0)",
+				this.indent([
+					"return Promise.resolve();"
+				]),
+				"",
+				"// a Promise means \"currently loading\".",
```

Comment:
Do the lint settings allow using single-quote or template (without inline replacements) to avoid escaping?